diff --git a/go/private/repositories.bzl b/go/private/repositories.bzl
index 991e6fc5..5a933fd0 100644
--- a/go/private/repositories.bzl
+++ b/go/private/repositories.bzl
@@ -173,6 +173,7 @@ def go_rules_dependencies(force = False):
         patches = [
             # releaser:patch-cmd gazelle -repo_root . -go_prefix github.com/golang/protobuf -go_naming_convention import_alias -proto disable_global
             Label("//third_party:com_github_golang_protobuf-gazelle.patch"),
+            Label("@com_github_buildbarn_bb_storage//:patches/com_github_golang_protobuf/service-registrar.diff"),
         ],
         patch_args = ["-p1"],
     )
@@ -277,6 +278,7 @@ def go_rules_dependencies(force = False):
         patches = [
             # releaser:patch-cmd gazelle -repo_root . -go_prefix github.com/golang/mock -go_naming_convention import_alias
             Label("//third_party:com_github_golang_mock-gazelle.patch"),
+            Label("@com_github_buildbarn_bb_storage//:patches/com_github_golang_mock/mocks-for-funcs.diff"),
         ],
         patch_args = ["-p1"],
         sha256 = "5359c78b0c1649cf7beb3b48ff8b1d1aaf0243b22ea4789aba94805280075d8e",
